### DISCUSSION TITLE
Is there a move to bar function?

### DISCUSSION THREAD

**Discussion by dabfalcon** [hydrogen-music/hydrogen#1719] [General]
Hey there Hydrogen,
I've been having a great time learning and using your software.
One thing I have not been able to work out from the documentation [(eg chapter 8](http://hydrogen-music.org/documentation/manual/manual_en.html) is how to move the position of the playhead to a specific bar. Is this possible?

I'm turning sheet music into a 'h2song' and I often find myself needing to trace up to see what bar I'm on in the song ruler, but sometimes the numbers are blank, or before, or after where I'd expect them to be. It would ideal if I could use a keyboard shortcut (like Ctrl + G in VS Code) to jump to a typed line / bar.

Any information regarding this or useful surrounding would be appreciated, thanks!

**Comment by theGreatWhiteShark** [comment]
Hey @dabfalcon,

You can use the [OSC API](http://hydrogen-music.org/documentation/manual/manual_en.html#chpt.osc.commands) `/Hydrogen/RELOCATE` to move to a specific column. Such an OSC command could be e.g. triggered by a commandline you open next to Hydrogen.

But that's just the column. If you are speaking about bar as in "beat, bar, and tick" things are a lot more tricky. Due to historical reasons Hydrogen does not properly support measures but just adds patterns of different length in subsequent columns. Using the JACK audio driver you can use another program, like Ardour, Muse, etc.,  as Timebase Master, set the bar within this program, and let the JACK server navigate Hydrogen to the specified bar. However, due to these historical incompatibilities the Timebase support is a bit experimental in Hydrogen and does only support different measure **or** tempo changes (See [BBT sync](http://hydrogen-music.org/documentation/manual/manual_en.html#chpt.preferences.audio_tab)).

**Reply by dabfalcon** [reply]
Great, thanks for the information :+1:
